pkg/systemd/hwinfo.jsx

```diff
@@ -24,61 +24,195 @@ import React from "react";
+        });
+    }
+
+    static availableMitigations() {
+        /* nosmt */
+        let promises = [cockpit.spawn(["lscpu"], { environ: ["LC_ALL=C.UTF-8"], }), cockpit.file("/proc/cmdline").read()];
+        return Promise.all(promises).then(values => {
+            let threads_per_core;
+            try {
+                threads_per_core = Number(values[0].split('\n').filter(l => l.indexOf('Thread(s) per core:') !== -1)[0].split(':')[1]);
+            } catch (e) {
+                console.warn(e);
+                return { available: false };
             }
-        </tbody>
-    </table>
-);
+            /* "nosmt" and "nosmt=force" are valid */
+            let nosmt_enabled = (values[1].indexOf("nosmt") !== -1 && values[1].indexOf("nosmt=") === -1) || values[1].indexOf("nosmt=force") !== -1;
+            /* available if threads>1 and the cmdline is valid */
+            let nosmt_available = (threads_per_core > 1 || (threads_per_core === 1 && nosmt_enabled)) && (values[1].indexOf("nosmt=") === -1 || values[1].indexOf("nosmt=force") !== -1);
+            return { available: nosmt_available, nosmt_enabled: nosmt_enabled };
+        });
+    }
+
+    close() {
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    saveAndReboot() {
+        let options = [];
+        if (this.state.nosmt)
+            options = ['set', 'nosmt'];
+        else
+            options = ['remove', 'nosmt'];
+        cockpit.script(kernelopt_sh, options, { superuser: "require" })
+                .then(() => {
+                    cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require" }).catch(error => this.setState({ rebooting: false, alert: error }));
+                })
+                .catch(error => this.setState({ rebooting: false, alert: error }));
+    }
+
+    render() {
+        let rows = [];
+        if (this.state.nosmt !== undefined)
+            rows.push((
+                <ListView.Item key="nosmt" heading={ <span>{ _("Disable symmetric multithreading") } (nosmt)</span> }
+                    actions={<div id="nosmt-switch"><OnOffSwitch disabled={this.state.rebooting} onChange={ value => this.setState({ nosmt: value }) } state={ this.state.nosmt } /></div> } >
+                    <span>{ _("Read more at: ") }<a role="link" href="https://access.redhat.com/security/vulnerabilities/L1TF">https://access.redhat.com/security/vulnerabilities/L1TF</a></span>
+                </ListView.Item>
+            ));
+
+        return (
+            <Modal show={this.props.show} id="cpu-mitigations-dialog">
+                <Modal.Header>
+                    <Modal.Title>{ _("CPU Security Toggles") }</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    { _("Software-based workarounds help prevent CPU security issues. These mitigations have the side effect of reducing performance. Turn on these settings at your own risk.") }
```

Comment:
This is semantically lopsided IMHO. Turning on a setting has a performance risk, while turning it off is a security risk. How about a more neutral "Change these settings at your own risk", or leaving out this sentence at all? @garrett